src/graph/LdapAuthenticator.h

```diff
@@ -0,0 +1,74 @@
+/* Copyright (c) 2020 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#ifndef GRAPH_LDAPAUTHENTICATOR_H_
+#define GRAPH_LDAPAUTHENTICATOR_H_
+
+#include "base/Base.h"
+#include "graph/Authenticator.h"
+#include <openldap.h>
+
+namespace nebula {
+namespace graph {
+
+class LdapAuthenticator final : public Authenticator {
+public:
+    /**
+     * Check if LDAP authentication parameters are set legally.
+     */
+    Status prepare();
+
+    /**
+     * Init LDAP connect to LADP server.
+     * FLAGS_ldap_server may be a space-separated list of hosts or IP addresses.
+     */
+    Status initLDAPConnection();
+
+    /**
+     * Build the filter
+     * Either search filter, or attribute filter, or default value.
+     */
+    std::string buildFilter();
+
+    /**
+     * Build the custom search filter.
+     * Replace all occurrences of the placeholder "$username" in FLAGS_ldap_searchfilter
+     * with variable userName_.
+     */
+    std::string buildSearchFilter();
+
+    /**
+     * Execute LDAP search bind mode authentication
+     */
+    StatusOr<bool> searchBindAuth();
+
+    /**
+     * Execute LDAP direct bind mode authentication
+     */
+    StatusOr<bool> directBindAuth();
+
+    /**
+     * Execute LDAP authentication.
+     */
+    StatusOr<bool> auth(const std::string &user, const std::string &password) override;
```

Comment:
I think , I think only one public method is enough of auth. other should be private. right?

Reply:
Yeah，you are right.